boards/arm/stm32f3_disco/doc/index.rst

```diff
@@ -210,9 +212,12 @@ Here is an example for the :ref:`hello_world` application.
 
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
-   :board: stm32f3_disco
+   :board: stm32f3_disco@E
    :goals: build flash
```

Comment:
Please let this part as is (default `stm32f3_disco` remains available) and add a new paragraph to introduce the new E revision